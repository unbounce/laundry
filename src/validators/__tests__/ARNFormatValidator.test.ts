import { lint } from '../../index';

describe('ARNFormatValidator', () => {
  test('valid', () => {
    const template = JSON.stringify({
      Resources: {
        Role: {
          Type: 'AWS::S3::Bucket',
          Properties: {
            BucketName: 'arn:aws:s3:::bar'
          }
        }
      }
    });
    expect(lint(template)).toEqual([]);
  });

  test('valid Ref', () => {
    const template = JSON.stringify({
      Parameters: {
        A: {
          Type: 'String',
          Default: 'arn:aws:s3:::bar'
        }
      },
      Resources: {
        Role: {
          Type: 'AWS::S3::Bucket',
          Properties: {
            BucketName: {
              Ref: 'A'
            }
          }
        }
      }
    });
    expect(lint(template)).toEqual([]);
  });

  test('invalid', () => {
    const template = JSON.stringify({
      Resources: {
        Role: {
          Type: 'AWS::S3::Bucket',
          Properties: {
            BucketName: 'arn:aws:s4:*:*:bar'
          }
        }
      }
    });
    expect(lint(template)).toMatchSnapshot();
  });

  test('invalid Ref', () => {
    const template = JSON.stringify({
      Parameters: {
        A: {
          Type: 'String',
          Default: 'arn:aws:s4:::bar'
        }
      },
      Resources: {
        Role: {
          Type: 'AWS::S3::Bucket',
          Properties: {
            BucketName: {
              Ref: 'A'
            }
          }
        }
      }
    });
    expect(lint(template)).toMatchSnapshot();
  });

});
