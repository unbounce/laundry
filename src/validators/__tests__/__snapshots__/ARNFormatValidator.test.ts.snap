// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ARNFormatValidator invalid 1`] = `
Array [
  Object {
    "message": "ARN service must one of a4b, apigateway, application-autoscaling, discovery, appstream, appsync, artifact, athena, autoscaling-plans, batch, aws-portal, budgets, acm, acm-pca, chime, cloud9, clouddirectory, cloudformation, cloudfront, cloudhsm, servicediscovery, cloudsearch, cloudtrail, cloudwatch, events, logs, codebuild, codecommit, codedeploy, codepipeline, signer, codestar, cognito-idp, cognito-identity, cognito-sync, comprehend, config, connect, cur, ce, datapipeline, dms, devicefarm, directconnect, ds, dynamodb, dax, autoscaling, ec2, ecr, ecs, eks, elasticbeanstalk, elasticfilesystem, elasticloadbalancing, elasticmapreduce, elastictranscoder, elasticache, es, fms, freertos, gamelift, glacier, globalaccelerator, glue, greengrass, guardduty, health, iam, importexport, inspector, iot, iotanalytics, iot1click, kms, kinesisanalytics, firehose, kinesis, kinesisvideo, lambda, lex, lightsail, macie, machinelearning, aws-marketplace, aws-marketplace-management, mechanicalturk, crowd, mediaconnect, mediaconvert, medialive, mediapackage, mediastore, mediatailor, ec2message, mgh, mobileanalytics, mobilehub, mq, opsworks, opsworks-cm, organizations, personalize, mobiletargeting, polly, pricing, quicksight, redshift, rekognition, rds, resource-groups, tag, route53, route53domains, route53resolver, sagemaker, secretsmanager, sts, serverlessrepo, servicecatalog, shield, shield, transfer, ses, sns, sqs, s3, swf, sdb, sso, snowball, states, storagegateway, sumerian, support, ssm, textract, transcribe, translate, trustedadvisor, ec2, waf, waf-regional, workdocs, worklink, workmail, workspaces, wam, xray, got s4, did you mean s3?",
    "path": Array [
      "Root",
      "Resources",
      "Role",
      "Properties",
      "BucketName",
    ],
    "source": "ARNFormatValidator",
  },
]
`;

exports[`ARNFormatValidator invalid Ref 1`] = `Array []`;
